File: grupa C/olimpiada/nacionalen/2009/2009-05-05-studentsC/submitsC/Log32002/trade.cpp
/*
TASK:trade
LANG:c++
*/
#include<iostream>
#include<cstdio>
#include<vector>
using namespace std;
int n,l=0;
double w[200000],q[200000];
struct edge
{
       int f;
       double fs;
       int s;
       double ss;
       };
       edge a[200000];
void read ()
{
     cin>>n;
     int i,x,y;
     double z;
     for(i=1;i<=n;i++)
     {
                      cin>>x>>y>>z;
                      if(z==0)l=1;
                      if(a[x].f==0){a[x].f=y;a[x].fs=z;}
                      else {a[x].s=y;a[x].fs=z;}
                      if(a[y].f==0){a[y].f=x;a[y].fs=z;}
                      else {a[y].s=x;a[y].fs=z;}
                      
     }
} 
void solve()
{
     int i;
     double dd;
     cout<<"impossible"<<endl;return;
     i=a[1].f;
     dd=a[i].fs;
     while(1)
     {
             if(a[i].s==1||a[i].f==1)break;
             else
             {
                if(a[i].s!=i){dd=a[i].ss-dd;i=a[i].s;}
                else {dd=a[i].fs-dd;i=a[i].f;} 
             }
     }
     w[1]=(a[1].ss-dd)/2;
     printf("%.2lf",w[1]);
}
int main ()
{
    read ();
    solve();
}

File: grupa C/olimpiada/nacionalen/2009/2009-05-05-studentsC/submitsC/Log32016/trade.pas
{
TASK:TRADE
LANG:PASCAL
}

program trade;
var n,i,j,aux,k:longint;
    st:real;
    lg:array[1..100000] of longint;
    sum,val:array[1..100000] of real;

function ok:boolean;
begin
  ok:=true;
  for i:= 1 to n do
    if abs(val[i]+val[lg[i]]-sum[i]-sum[i])>0.001 then
      ok:=false;
end;

procedure afla;
var semn:byte;
    nxt:longint;
begin
  for i:= 1 to n do
    begin
      semn:=0;
      nxt:=i;
      repeat
        if semn =0 then
          val[i]:=val[i]+sum[nxt]
          else val[i]:=val[i]-sum[nxt];
        nxt:=lg[nxt];
        semn:=not semn;
      until nxt=i;
      if n mod 2 =0 then
        val[i]:=val[i]*-1;
    end;
end;

begin
 { assign(input,'trade.in'); reset(input);
  assign(output,'trade.out'); rewrite(output);  }
  readln(n);
  for k:=1 to n do
    begin
      read(i,j);
      if i>j then
       begin
         aux:=j;
         j:=i;
         i:=aux;
       end;
      if lg[i]<> 0 then
        begin
          lg[j]:=i;
          read(sum[j]);
        end
        else
          begin
            lg[i]:=j;
            read(sum[i]);
          end;

    end;
  afla;
  if not ok then
    begin
      writeln('impossible');
    end
    else
      begin
        for i:= 1 to n do
          write(val[i]:0:2,' ');
        writeln;
      end;

 { close(output);
  close(input); }
end.